internal/ethapi/api.go

```diff
@@ -861,7 +861,10 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	if evm.Cancelled() {
 		return nil, fmt.Errorf("execution aborted (timeout = %v)", timeout)
 	}
-	return result, err
+	if err != nil {
+		return result, fmt.Errorf("err: %w (supplied gas %d)", err, msg.Gas())
```

Comment:
Any particular reason to include the assigned gas to error message?

Reply:
When I tested it, I noticed that it's very UX-unfriendly to say "err: insufficient gas", when the user has provided `10M`. The user will keep increasing it and scratching his head as to why it doesn't work. So it's nicer to somehow signal that it doesn't matter what he passes, it's always capped.